components/measurement/useTestKeyController.js

```diff
@@ -0,0 +1,71 @@
+import React, { useState, useCallback } from 'react'
+import { Flex } from 'ooni-components'
+
+// Custom hook to render a toolbar to enable/disable parts of `testKeys`
+// Usage: Add this to the MeasurementContainer
+/*
+import { useTestKeyController } from './useTestKeyController'
+...
+...
+const { testKeys, TestKeyController } = useTestKeyController(measurement.test_keys)
+const measurementMod = Object.assign({}, measurement, { test_keys: testKeys })
+
+return (
+  <React.Fragment>
+    <TestKeyController />
+    <TestDetails measurement={measurementMod} {...props} />
+  </React.Fragment>
+)
+*/
+
+const setValues = (input, value = true) => {
+  return Object.keys(input).reduce((o, k) => {
+    o[k] = value
+    return o
+  }, {all: value})
+}
+
+export const useTestKeyController = (testKeysInitial) => {
+  const [testKeys, setTestKeys] = useState(testKeysInitial)
+  const [keysMap, setKeysMap] = useState(setValues(testKeysInitial))
+
+  const TestKeyController = () => {
+    const onChange = useCallback((event) => {
+      const {name, checked} = event.target
+
+      if (name === 'all') {
+        setTestKeys(checked ? testKeysInitial : {})
+        setKeysMap(setValues(testKeysInitial, checked))
+        return
+      } else {
```

Comment:
I'd like to encourage you to avoid `if () { ... return } else { ... }`. Why don't we follow instead the Okkam razor's principle where we don't include an else branch when we don't need it?

Reply:
Thanks. You are right. I have now changed it to a simple exclusive `if ... then` structure. Earlier I wrote some common steps at the end, which eventually changed.